src/main/java/net/mcreator/generator/LocalizationUtils.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2020 Pylo and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.generator;
+
+import com.google.gson.GsonBuilder;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.generator.template.TemplateExpressionParser;
+import net.mcreator.io.FileIO;
+import net.mcreator.workspace.Workspace;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LocalizationUtils {
+
+	public static void generateLanguageFiles(Generator generator, Workspace workspace, Map<?, ?> config) {
+		if (config.size() > 0)
+			switch ((String) config.get("format")) {
+			case "keyvalue":
+				generateKeyValueLanguageFiles(generator, workspace, config);
+				break;
+			case "json":
+				generateJSONLanguageFiles(generator, workspace, config);
+				break;
+			default:
+				break;
+			}
+	}
+
+	private static void generateKeyValueLanguageFiles(Generator generator, Workspace workspace, Map<?, ?> config) {
+		String rawName = (String) config.get("langfile_name");
+
+		if (!workspace.getFolderManager().isFileInWorkspace(generator.getLangFilesRoot()))

```

Comment:
In what cases may localization files get located outside the current workspace? If the mod type does not support localization? Anything else?

Reply:
If plugin tries to escape and generate file somewhere else on the computer